File: snippets/fsharp/System/Type/FullName/Fullname3.fs
module Fullname3x

// <Snippet3>
open System

let t = typeof<Nullable<_>>.GetGenericTypeDefinition()
printfn $"{t.FullName}"
if t.IsGenericType then
    printf "   Generic Type Parameters: "
    let gtArgs = t.GetGenericArguments()
    for arg in gtArgs do
        match arg.FullName with
        | null -> "(unassigned) " + string arg
        | _ -> arg.FullName
        |> printfn "%s"
    printfn ""
// The example displays the following output:
//       System.Nullable`1
//          Generic Type Parameters: (unassigned) T
// </Snippet3>

File: snippets/csharp/System/Type/FullName/Fullname3.cs
// <Snippet3>
using System;


public class FullName3Example
{
    public static void Run()
    {
        Type t = typeof(Nullable<>);
        Console.WriteLine(t.FullName);
        if (t.IsGenericType)
        {
            Console.Write("   Generic Type Parameters: ");
            Type[] gtArgs = t.GetGenericArguments();
            for (int ctr = 0; ctr < gtArgs.Length; ctr++)
            {
                Console.WriteLine(gtArgs[ctr].FullName ??
                                  "(unassigned) " + gtArgs[ctr]);
            }
            Console.WriteLine();
        }
    }
}
// The example displays the following output:
//       System.Nullable`1
//          Generic Type Parameters: (unassigned) T
// </Snippet3>
